### PR TITLE
System monitoring manual update

### PR DESCRIPTION
**Pull request by sanchitanec**

### CODE REVIEW

File: docker/core/http/docker-compose.yml [hunk 1]
```diff
@@ -104,3 +104,34 @@ services:
...
+        image: fogflow/metricbeat:7.6.0
+        user: root
+        volumes:
+          - ./metricbeat.docker.yml:/usr/share/metricbeat/metricbeat.yml
+          - /var/run/docker.sock:/var/run/docker.sock
+          - /sys/fs/cgroup:/hostfs/sys/fs/cgroup
+          - /proc:/hostfs/proc
+          - /:/hostfs
+        environment:
+          - output.elasticsearch.hosts= ["<Cloud_Public_IP>:9200"]
```
Comment (ajayyadavnec) [inline docker/core/http/docker-compose.yml]: check the docker policy is it required restart(restart: always) the container at start or exit like other  containers
refer https://stackoverflow.com/questions/41555884/docker-what-does-docker-run-restart-always-actually-do

File: docker/core/http/docker-compose.yml [hunk 2]
```diff
@@ -104,3 +104,34 @@ services:
...
+#          - network.host= "0.0.0.0"
+    metricbeat:
+        image: fogflow/metricbeat:7.6.0
+        user: root
+        volumes:
+          - ./metricbeat.docker.yml:/usr/share/metricbeat/metricbeat.yml
+          - /var/run/docker.sock:/var/run/docker.sock
+          - /sys/fs/cgroup:/hostfs/sys/fs/cgroup
+          - /proc:/hostfs/proc
+          - /:/hostfs
```
Comment (ajayyadavnec) [inline docker/core/http/docker-compose.yml]: check why we do not require to expose port for metrics container

File: doc/en/source/edge.rst [hunk 3]
```diff
@@ -75,6 +76,30 @@ You can use the default setting for a simple test, but you need to change the fo
 	
     } 
 
+Change the Metricbeat configuration file accordingly
+-----------------------------------------------------------
+
+you need to change the following addresses in start.sh according to your own environment:
+-**output.elasticsearch.hosts**: It is the elasticsearch host IP adress on which metricbeat will share the metric data.
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/edge.rst]: spell corrected

[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/edge.rst]: corrected the formatting

File: doc/en/source/edge.rst [hunk 4]
```diff
@@ -75,6 +76,30 @@ You can use the default setting for a simple test, but you need to change the fo
...
+
+        name: "<155.54.239.141/edge02>"
+        metricbeat.modules:
+        - module: docker
+          metricsets: ["cpu","memory","network"]
+          hosts: ["unix:///var/run/docker.sock"]
+          period: 10s
+          enabled: true
+        - module: system
+          metricsets: ["cpu","load","memory","network"]
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/edge.rst]: Added the comment before parameters for the same.

File: doc/en/source/onepage.rst [hunk 5]
```diff
@@ -8,7 +7,21 @@ In the FIWARE-based architecture, FogFlow can be used to dynamically trigger dat
 between IoT devices and Orion Context Broker, 
 for the purpose of transforming and preprocessing raw data at edge nodes (e.g., IoT gateways or Raspberry Pis).
 
-The tutorial introduces a typical FogFlow system setup with a simple example to do anomaly detection at edges for temperature sensor data. 
+To monitor system resource usage either on cloud node or on edge node can be done with the help of system monitoring tools.
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: This has been discussed Bin in meeting, also discussed in status meeting.

File: doc/en/source/onepage.rst [hunk 6]
```diff
@@ -8,7 +7,21 @@ In the FIWARE-based architecture, FogFlow can be used to dynamically trigger dat
...
+Here in FogFlow we use Metricbeat, Elasticsearch and Grafana in short EMG. With this set of tools we can monitor the FogFlow edge nodes
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: Removed active voice

File: doc/en/source/onepage.rst [hunk 7]
```diff
@@ -8,7 +7,21 @@ In the FIWARE-based architecture, FogFlow can be used to dynamically trigger dat
 between IoT devices and Orion Context Broker, 
 for the purpose of transforming and preprocessing raw data at edge nodes (e.g., IoT gateways or Raspberry Pis).
 
-The tutorial introduces a typical FogFlow system setup with a simple example to do anomaly detection at edges for temperature sensor data. 
+To monitor system resource usage either on cloud node or on edge node can be done with the help of system monitoring tools.
+Here in FogFlow we use Metricbeat, Elasticsearch and Grafana in short EMG. With this set of tools we can monitor the FogFlow edge nodes 
+that join with FogFlow cloud node on the fly.
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: Replaced

File: doc/en/source/onepage.rst [hunk 8]
```diff
@@ -114,38 +156,97 @@ There are two ways to check if the FogFlow cloud node is started correctly:
...
+	bb8e25e5a75d      fogflow/broker          "/broker"                9 seconds ago       Up 7 seconds        0.0.0.0:8070->8070/tcp                               fogflow_cloud_broker_1
+	7f3ce330c204      rabbitmq:3              "docker-entrypoint.s…"   10 seconds ago      Up 6 seconds        4369/tcp, 5671/tcp, 25672/tcp, 0.0.0.0:5672->5672/tcp     fogflow_rabbitmq_1
+	9e95c55a1eb7      fogflow/discovery       "/discovery"             10 seconds ago      Up 8 seconds        0.0.0.0:8090->8090/tcp                               fogflow_discovery_1
+        399958d8d88a      grafana/grafana:6.5.0   "/run.sh"                29 seconds ago      Up 27 seconds       0.0.0.0:3003->3000/tcp                               fogflow_grafana_1
+        9f99315a1a1d      fogflow/elasticsearch:7.5.1 "/usr/local/bin/dock…" 32 seconds ago    Up 29 seconds       0.0.0.0:9200->9200/tcp, 0.0.0.0:9300->9300/tcp       fogflow_elasticsearch_1
+        57eac616a67e      fogflow/metricbeat:7.6.0 "/usr/local/bin/dock…"   32 seconds ago     Up 29 seconds                                                                  fogflow_metricbeat_1
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: There is no requirement to put/expose ports in metricbeat

File: doc/en/source/onepage.rst [hunk 9]
```diff
@@ -114,38 +156,97 @@ There are two ways to check if the FogFlow cloud node is started correctly:
...
+	
 
 Once you are able to access the FogFlow dashboard, you can see the following web page
 
+
 .. figure:: figures/dashboard.png
 
 
+
+Configure Elasticsearch on Grafana Dashboard
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: Reason same as in review comment#4.

File: doc/en/source/onepage.rst [hunk 10]
```diff
@@ -114,38 +156,97 @@ There are two ways to check if the FogFlow cloud node is started correctly:
...
+-------------------------------------------------------------
+
+You can open the Grafana dashboard in your web browser to see the current system status via the URL: 
+http://<output.elasticsearch.hosts>:3003/. The default username and password for grafana login are admin and admin respectively.
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: Done

File: doc/en/source/onepage.rst [hunk 11]
```diff
@@ -114,38 +156,97 @@ There are two ways to check if the FogFlow cloud node is started correctly:
...
+
+
+- After successful login to grafana, click on "Create your first data source" on Home Dashboard to setup the source of data.
+- Select Elasticsearch from Add Data Sourch page. Now you are on page Data Sources/Elasticsearch same as below figure.
+
+
+.. figure:: figures/Elastic_config.png
+
+
+1. put a name for the Data Source.
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: Done

File: doc/en/source/onepage.rst [hunk 12]
```diff
@@ -114,38 +156,97 @@ There are two ways to check if the FogFlow cloud node is started correctly:
...
+2. In HTTP detail ,mention URL of your elasticsearch and Port. URL shall include http.
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply doc/en/source/onepage.rst]: Done

File: docker/core/http/docker-compose.yml [hunk 13]
```diff
@@ -104,3 +104,34 @@ services:
           - IOTA_HTTP_PORT=7896 # The port used for device traffic over HTTP
           - IOTA_PROVIDER_URL=http://<Cloud_Public_IP>:4041 # URL for the IoT Agent to be deployed
           - IOTA_DEFAULT_RESOURCE=/iot/json
+
+    elasticsearch:
+        image: fogflow/elasticsearch:7.5.1
```
[the remark this answers is not in the document]
Comment (sanchitanec) [reply docker/core/http/docker-compose.yml]: Done, Now images can be pulled with latest tag.

**Comment by sanchitanec** [comment]
Comments are incorporated

**Pull request merged**